### ISSUE TITLE
Seeing multiple definitions when hovering over a symbol

### ISSUE THREAD

**Issue by moodmosaic** [SublimeHaskell/SublimeHaskell#311]
Is this behavior correct? It looks like it's giving me all possible definitions it can find (?)

![image](https://cloud.githubusercontent.com/assets/287532/24071847/5035f70e-0be4-11e7-8253-86b89638d1bd.png)

---

Here's another example of this behavior, where the same type info is shown twice (!)

![image](https://cloud.githubusercontent.com/assets/287532/24071853/7429af8e-0be4-11e7-86a3-9d23eb70f9fa.png)

(Notice also that `searchTag` on line 67 should be blue and not white, but that's for a different issue...)

**Comment by mvoidex** [comment]
Yes, first one is inferred type, second is generic. Maybe message could better to make it clear.

**Comment by moodmosaic** [comment]
Hmm... What's happens if there are more than one generic ones? It'll show a huge list?

**Comment by mvoidex** [comment]
By generic I mean type signature (i.e. function was declared with), so it should be only one

**Comment by moodmosaic** [comment]
I still have a hard time to figure this out :( If you look at the 1st screenshot, the first type signature looks different than the second one.

```
try :: ParsecT String ...

try :: Exception e ...
```

**Comment by mvoidex** [comment]
Oh, I see, you're right. It's bug in name lookup - it shows two different `try`, and correct one is first. I'm experimenting with using `haskell-names` package for names lookup, but there are still some work to be done before it will be ready to merge.

**Comment by moodmosaic** [comment]
In case it helps, this seems to be happening only on Windows... I don't see it happening on Ubuntu.

---

In general, hsdev/SublimeHaskell is way more stable and faster on Ubuntu...
